### ISSUE TITLE
Large number of calls to blob storage by EventProcessorHost for lease Mgmt

### ISSUE THREAD

**Issue by stephenschwan** [Azure/azure-event-hubs-dotnet#331]
Behavior is consistent in both our environments, example data for each is as follows:

- All 3 event hubs use the same storage account for blob mgmt.  
- Default settings are being used on the ProcessorHost setups.
- Processor hosts are running on two instances, 4 core azure app services

**Non-Prod:**
-EventHub1: 1 consumer group, 4 partitions
-EventHub2: 4 consumer groups, 5 partitions
-EventHub3: 2 consumer groups, 4 partitions

On average, with this setup, it is making 10K requests / min

**Prod:**
-EventHub1: 1 consumer group, 32 partitions
-EventHub2: 4 consumer groups, 32 partitions
-EventHub3: 2 consumer groups, 32 partitions

On average, with this setup, it is making 20K requests / min for the region getting traffic.  Our secondary with same setup with no traffic averages 10K request/min

_EDIT: Updated numbers above - had the wrong aggregation bucket initially (5min vs 1min)_

So, is this expected behavior?  Should folks not use the default settings for ProcessorHost?  We can use my non-prod setup to verify / validate to find answers etc...

Below is the chart for blob transactions.  At the end, I shut down our processors and you can see the # of calls drops off.
![non-prod-calls](https://user-images.githubusercontent.com/9887409/47042515-504aed80-d140-11e8-8292-cb7294c2822b.JPG)

## Versions  
- OS platform and version: Azure App Service
- .NET Version: .net core 2.1, standard 2.0
- NuGet package version or commit ID: 
    - Microsoft.Azure.EventHubs.Processor 2.2.1
    - Microsoft.Azure.EventHubs 2.2.1

**Comment by serkantkaraca** [comment]
I am running a sample with 32 partitions and 1 host right now. I will share my results.

**Comment by serkantkaraca** [comment]
W/o any checkpointing, my test resulted in 400 storage requests per minute with 32 partitions on a single host. While checking the code, I have identified a small room to lower storage requests. PR - https://github.com/Azure/azure-event-hubs-dotnet/pull/333

I think in your case, most of the storage requests are originated by checkpointing. Can you do a test run w/o call checkpoint?

**Comment by serkantkaraca** [comment]
I am working on another changeset to reduce storage calls during checkpointing. I will put that changes under test and see if anything is broken.

**Comment by stephenschwan** [comment]
So, I am pretty sure this is me.  We did change the default batch size in our event processors to 100.  I think the default is 300.  So, in theory, we would have 3 times the calls as default settings.  

I did a test on our side with default setup, and the numbers aligned to what you published.  I think we can close the issue now as our checkpointing is causing the higher numbers.

thanks,
stephen
